src/serialization/sb2.js

```diff
@@ -423,9 +444,54 @@ const parseScratchObject = function (object, runtime, extensions, topLevel, zip)
         }
     }
 
+    // If included, parse any and all comments on the object (this includes top-level
+    // workspace comments as well as comments attached to specific blocks)
+    const blockComments = {};
+    if (object.hasOwnProperty('scriptComments')) {
+        const comments = object.scriptComments.map(commentDesc => {
+            const isBlockComment = commentDesc[5] >= 0;
+            const newComment = new Comment(
+                null, // generate a new id for this comment
+                commentDesc[6], // text content of sb2 comment
+                // Only serialize x & y position of comment if it's a workspace comment
+                // If it's a block comment, we'll let scratch-blocks handle positioning
+                isBlockComment ? null : commentDesc[0] * 1.5, // x position of comment
```

Comment:
Where do these 1.5 and 2.2 numbers come from? Maybe they could use an explanatory variable name instead?

Reply:
I found them as [existing magic numbers](https://github.com/LLK/scratch-vm/blob/develop/src/serialization/sb2.js#L153-L154) to translate block coordinates. =]

I'll make constants and use them in both places.

Reply:
🌮